extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/AssetApiController.java

```diff
@@ -67,7 +68,7 @@ public void createAsset(@Valid AssetEntryDto assetEntryDto) {
         var dataAddressResult = transformerRegistry.transform(assetEntryDto.getDataAddress(), DataAddress.class);
 
         if (assetResult.failed() || dataAddressResult.failed()) {
-            throw new IllegalArgumentException("Request is not well formatted");
+            throw new InvalidRequestException("Request is not well formatted");
```

Comment:
We're using this message string in several locations, maybe we could add a static method like `throw InvalidRequestException.requestFormat()` to have the message text in a central place?

Reply:
it's used 3 times and in two of them there's an additional detail. I agree that we should avoid this kind of duplication but I don't see a real advantage here.

Reply:
could it be a `.requestFormat(@Nullable String additionalDetail)`, or an override?

Reply:
@paullatzelsperger actually this reminded me that I had an idea, provide a list of messages to the `EdcApiException` so there could be a better mapping from `Result` to `EdcApiException` to a `List<ApiErrorDetail>`, this will make the exception accepting a list of messages from the failure, without having to specify any further message (if I see a 400 error it's pretty obvious that the "Request is not well formatted").